x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferencePlugin.java

```diff
@@ -208,11 +214,22 @@ public List<InferenceServiceExtension.Factory> getInferenceServiceFactories() {
 
     @Override
     public List<NamedWriteableRegistry.Entry> getNamedWriteables() {
-        var entries = new ArrayList<NamedWriteableRegistry.Entry>();
-        entries.addAll(InferenceNamedWriteablesProvider.getNamedWriteables());
+        var entries = new ArrayList<>(InferenceNamedWriteablesProvider.getNamedWriteables());
+        entries.add(new NamedWriteableRegistry.Entry(RankBuilder.class, TextSimilarityRankBuilder.NAME, TextSimilarityRankBuilder::new));
         return entries;
     }
 
+    @Override
+    public List<NamedXContentRegistry.Entry> getNamedXContent() {
+        return List.of(
+            new NamedXContentRegistry.Entry(
+                RankBuilder.class,
+                new ParseField(TextSimilarityRankBuilder.NAME),
+                TextSimilarityRankBuilder::fromXContent
+            )
+        );
+    }
+
```

Comment:
I am confused on the difference between a "RankBuilder" and a "RetrieverBuilder"

The retriever spec registration adds named xcontent automatically via
```
    private void registerRetriever(RetrieverSpec<?> spec) {
```

@pmpailis does that not work anylonger?

Reply:
For parsing, I think that just the export through the parent `TextSimilarityRankRetrieverBuilder` [here](https://github.com/elastic/elasticsearch/blob/f1155a554d885e010092225b86887947c4e30c76/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferencePlugin.java#L329), should be sufficient enough, as we don't expect to parse directly any XContent for the  `RankBuilder`. We do need however the `NamedWriteable`  as we pass the `RankBuilder` through `source`. 

Also, just for completeness, a `RetrieverBuilder` is the top-level element in the tree and is one of `knn`, `standard`, `rrf`, and now, `text-similarity-rank`. The `RankBuilder` on the other hand is something specific that only `rrf` and `text-similarity-rank` retrievers specify and export to source, and defines the necessary steps to apply any custom ranking. So eventually, as the retriever will be flatten out to be added to source, the `rankBuilder` will be left as a primary object of `source` that needs to be (de)serialized.  

It's quite confusing and the concepts are tied together in a not-so-clear way, but hopefully this will be much more clear with the pending rework and clear distinction of the `retriever` and `ranker` concepts.